lib/closex/caching_client.ex

```diff
@@ -14,14 +14,36 @@ defmodule Closex.CachingClient do
   end
 
   defdelegate create_lead(payload, opts \\ []), to: @fallback_client
-  defdelegate update_lead(lead_id, payload, opts \\ []), to: @fallback_client
+
+  def update_lead(lead_id, payload, opts \\ []) do
+    case apply(@fallback_client, :update_lead, [lead_id, payload, opts]) do
+      {:ok, updated_lead} ->
+        case set_cached(lead_id, updated_lead) do
```

Comment:
I would not set this cached value from the update, since close.io might reject it for whatever reason. Instead, we should simply invalidate the cache for this key.